src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/components/GenericComponent.tsx

```diff
@@ -145,12 +138,7 @@ export function GenericComponent(props: IGenericComponentProps) {
       return;
     }
 
-    if (formData instanceof Object) {
-      if (formData[key] && formData[key] === value) {
-        // data unchanged, do nothing
-        return;
-      }
-    } else if (formData && formData === value) {
+    if (formData[key] === value) {
```

Comment:
```suggestion
    if (formData[key] && formData[key] === value) {
```
@jeevananthank This is the offending change.

@haakemon I'm tempted to just remove the whole equality check. As far as I can tell it is only for performance reasons and as components only update redux state onBlur (or onClick) the check will only improve performance when tabbing trough lots of fields. Those will most likely be empty fields, that apparently should be validated.

Reply:
@jeevananthank What is expected behaviour here
* Open a form with a max length restriction on an input field
* Type a too long value (get the error)
* Refresh page (no error displayed)
* Tab trough all the fields

Should the error be displayed now? (or is it just errors on empty fields that should error after tabbing trough without changing?)

Reply:
![image](https://user-images.githubusercontent.com/131616/148911593-9007f5ff-304f-4a1b-af85-1487704cc06b.png)

Reply:
I believe the expected behavior should be that the error should be displayed on tabbing through fields with unexpected values. @Febakke can you comment/confirm on this ?

Reply:
Removing the check might work, but I'm not confident in making such a change really. Are we sure the test coverage is good enough around this area that we dont accidentally change behavior, etc? Personally I have too little insight into altinn-app-frontend project and how it works. Perhaps @lorang92 has some thoughts on this, but he's not back until later this week.

Reply:
This issue was merged as part of #7718, so I'll resolve this conversation even though I think it is a bug to only trigger validation on unchanged fields if the field is empty.